File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        freq = {}

        for c in s:
            if c not in freq:
                freq[c] = 1
            else:
                freq[c] +=1

        for c in t:
            if c in freq:
                freq[c] -=1
            else:
                return False

        for value in freq.values():
            if value != 0:
                return False

        return True                           




        
        

File: Data Structures & Algorithms/majority-element-ii/submission-2.cpp
#include<unordered_map>
using namespace std;
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int> result;
        unordered_map<int,int> freq;
        const int size = nums.size();
        const int amount = size/3;

        for(int i: nums){
            freq[i] ++;
        }

        for(auto &[key, value]: freq){
            if(value>amount){
                result.push_back(key);
            }

        }
        return result;
    }
};